src/azure/cli/commands/arm.py

```diff
@@ -150,3 +151,201 @@ def required_values_validator(namespace):
+
+    get_arguments = dict(extract_args_from_signature(getter))
+    set_arguments = dict(extract_args_from_signature(setter))
+
+    def handler(args):
+        ordered_arguments = args.pop('ordered_arguments')
+
+        getterargs = {key: val for key, val in args.items()
+                      if key in get_arguments}
+        instance = getter(**getterargs)
+
+        # Update properties
+        for arg in ordered_arguments:
+            arg_type, expressions = arg
+            if arg_type == '--set':
+                try:
+                    for expression in expressions:
+                        set_properties(instance, expression)
+                except ValueError:
+                    raise CLIError('--set should be of the form:'
+                                   ' --set property.property=<value>'
+                                   ' property2.property=<value>')
+            elif arg_type == '--add':
+                try:
+                    add_properties(instance, expressions)
+                except ValueError:
+                    raise CLIError('--add should be of the form:'
+                                   ' --add property.list key1=value1 key2=value2')
+            elif arg_type == '--remove':
+                try:
+                    remove_properties(instance, expressions)
+                except ValueError:
+                    raise CLIError('--remove should be of the form: --remove'
+                                   ' property.propertyToRemove or'
+                                   ' --remove property.list <indexToRemove>')
+            else:
+                raise ValueError('Unsupported arg type {}'.format(arg_type))
+
+        # Done... update the instance!
+        getterargs[setter_arg_name] = instance
+        opres = setter(**getterargs)
+        return opres.result() if isinstance(opres, AzureOperationPoller) else opres
+
+    class OrderedArgsAction(argparse.Action): #pylint:disable=too-few-public-methods
+        def __call__(self, parser, namespace, values, option_string=None):
+            if not getattr(namespace, 'ordered_arguments', None):
+                setattr(namespace, 'ordered_arguments', [])
+            namespace.ordered_arguments.append((option_string, values))
+
+    def one_required(namespace):
+        if not getattr(namespace, 'ordered_arguments', None):
+            raise ValueError('At least one must be specified: --add, --set, --remove')
+
+    cmd = CliCommand(name, handler)
+    cmd.arguments.update(set_arguments)
+    cmd.arguments.update(get_arguments)
+    cmd.arguments.pop(setter_arg_name, None)
+    cmd.add_argument('properties_to_set', '--set', nargs='+', action=OrderedArgsAction, default=[],
+                     help='Update an object by specifying a property path and value to set.'
+                     '  Example: --set property1.property2=value',
+                     metavar='PROPERTY1.PROPERTY2=VALUE')
```

Comment:
> metavar='PROPERTY1.PROPERTY2=VALUE') [](start = 20, length = 37)

KEY=VALUE? 

The (minor) issue here is that the usage strings becomes very long with long metavars - especially for nargs commands, and that makes it hard to read...